packages/monitor-v2/src/llm-bot/common.ts

```diff
@@ -124,42 +138,29 @@ export abstract class OptimisticOracleClient {
   getFetchedBlockRange(): [number, number] {
     return this.fetchedBlockRange;
   }
-}
-
-export class OptimisticOracleClientV2 extends OptimisticOracleClient {
-  constructor(_provider: Provider, _requests: OptimisticOracleRequest[] = [], _fetchedBlockRange?: [number, number]) {
-    super(_provider, _requests, _fetchedBlockRange);
-  }
-
-  protected async fetchOracleRequests(blockRange: [number, number]): Promise<OptimisticOracleRequest[]> {
-    // TODO: Implement this for the OptimisticOracleV2
-    blockRange;
-    return [];
-  }
 
-  protected createClientInstance(
-    requests: OptimisticOracleRequest[],
-    fetchedBlockRange: [number, number]
-  ): OptimisticOracleClient {
-    return new OptimisticOracleClientV2(this.provider, requests, fetchedBlockRange);
+  /**
+   * Returns the provider used for interacting with the blockchain.
+   * @returns The provider object.
+   */
+  getProvider(): Provider {
+    return this.provider;
   }
 }
 
-export class OptimisticOracleClientV3 extends OptimisticOracleClient {
-  constructor(_provider: Provider, _requests: OptimisticOracleRequest[] = [], _fetchedBlockRange?: [number, number]) {
-    super(_provider, _requests, _fetchedBlockRange);
-  }
-
-  protected async fetchOracleRequests(blockRange: [number, number]): Promise<OptimisticOracleRequest[]> {
-    // TODO: Implement this for the OptimisticOracleV3
-    blockRange;
-    return [];
-  }
-
-  protected createClientInstance(
-    requests: OptimisticOracleRequest[],
-    fetchedBlockRange: [number, number]
-  ): OptimisticOracleClient {
-    return new OptimisticOracleClientV2(this.provider, requests, fetchedBlockRange);
-  }
+/**
+ * Represents a filtering strategy for an Optimistic Oracle client.
+ * @template I The type of the input Optimistic Oracle client.
+ * @template O The type of the output Optimistic Oracle client.
+ */
+export interface OptimisticOracleClientFilter<
+  I extends OptimisticOracleClient<OptimisticOracleRequest>,
+  O extends OptimisticOracleClient<OptimisticOracleRequest>
+> {
+  /**
+   * Filters the input Optimistic Oracle client and returns the filtered output Optimistic Oracle client.
+   * @param optimisticOracleClient The input Optimistic Oracle client to be filtered.
+   * @returns A Promise that resolves to the filtered output Optimistic Oracle client.
+   */
+  filter(optimisticOracleClient: I): Promise<O>;
```

Comment:
Just curious: why did you decide to have the filter take in an entire client rather than just a list of requests?

Reply:
Hey @mrice32 
I though passing the client to the LLMStrategy class could make sense in case the client carried any useful information or function but thinking more about it I thing you are right. I've changed the filter interface to work generic `OptimisticOracleRequest`s

Reply:
Awesome!